compose/snippets/src/main/java/com/example/compose/snippets/text/AutofillSnippets.kt

```diff
@@ -0,0 +1,138 @@
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalAutofillManager
+import androidx.compose.ui.semantics.contentType
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import com.example.compose.snippets.touchinput.Button
+
+@Composable
+fun AddAutofill() {
+    // [START android_compose_autofill_1]
+    BasicTextField(
+        state = remember { TextFieldState("Enter your username.") },
+        modifier = Modifier.semantics { contentType = ContentType.Username }
+    )
+    // [END android_compose_autofill_1]
+}
+
+@Composable
+fun AddMultipleTypesOfAutofill() {
+    // [START android_compose_autofill_2]
+    Column {
+        BasicTextField(
+            state = remember { TextFieldState() },
+            modifier =
+            Modifier.semantics {
+                contentType = ContentType.Username + ContentType.EmailAddress
+            },
+        )
+    }
+    // [END android_compose_autofill_2]
+}
+
+@Composable
+fun SaveDataWithAutofill() {
+    // [START android_compose_autofill_3]
+    // [START android_compose_autofill_4]
+    val autofillManager = LocalAutofillManager.current
+    // [END android_compose_autofill_3]
+
+    Column {
+        BasicTextField(
+            state = remember { TextFieldState() },
+            modifier =
+            Modifier.semantics {
+                contentType = ContentType.NewUsername
+            },
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        BasicTextField(
+            state = remember { TextFieldState() },
+            modifier =
+            Modifier.semantics {
+                contentType = ContentType.NewPassword
+            },
+        )
+    }
+    // [END android_compose_autofill_4]
+}
+
+@Composable
+fun SaveDataWithAutofillOnClick() {
+    // [START android_compose_autofill_5]
+    val autofillManager = LocalAutofillManager.current
+    Column {
+        BasicTextField(
+            state = remember { TextFieldState() },
+            modifier =
+            Modifier.semantics {
+                contentType = ContentType.NewUsername
+            },
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        BasicTextField(
+            state = remember { TextFieldState() },
+            modifier =
+            Modifier.semantics {
+                contentType = ContentType.NewPassword
+            },
+        )
+
+        // Submit button
+        Button(onClick = { autofillManager?.commit() }) { Text("Reset credentials") }
+    }
+    // [END android_compose_autofill_5]
+}
+
+// [START android_compose_autofill_6]
+@Composable
+fun customizeAutofillHighlight() {
+    val customHighlightColor = Color.Red
+    val usernameState = remember { TextFieldState() }
+
+    CompositionLocalProvider(LocalAutofillHighlightColor provides customHighlightColor) {
+        Column {
+            BasicTextField(
+                state = usernameState,
```

Comment:
For consistency, should we just get rid of the `usernameState` val and replace inline on line 130 `state = remember ( TextFieldState() }` like the other samples do?

Reply:
yeah i'll update